### PR TITLE
stm32_common: unified RTC driver

### PR DESCRIPTION
**Pull request by vincent-d**
Hi,

This PR unify the stm32f0, f2, f3, f4 and l1 RTC driver.

The init behavior is changed a little bit to avoid resetting the counter if the RTC was already enabled.

According to this: http://www.st.com/content/ccc/resource/technical/document/application_note/7a/9c/de/da/84/e7/47/8a/DM00025071.pdf/files/DM00025071.pdf/jcr:content/translations/en.DM00025071.pdf
all these families share (almost) the same RTC.

Tested on f0 and f2 with tests/periph_rtc and default example. Compiled on f3, f4 and l1 (I modified some board to compile the RTC code) but as I don't have any board with these mcu I can't test.

### CODE REVIEW

**Comment by aabadie** [comment]
Nice !

File: cpu/stm32_common/periph/rtc.c [hunk 1]
```diff
@@ -1,16 +1,21 @@
 /*
- * Copyright (C) 2016 OTA keys S.A.
+ * Copyright (C) 2015 Lari Lehtomäki
+ *               2016 Laksh Bhatia
+ *               2016 OTA keys S.A.
```
Comment (aabadie) [inline cpu/stm32_common/periph/rtc.c]: Maybe 2016-2017 for OTA keys ?

**Comment by aabadie** [comment]
Tested tests/periph_rtc on a few boards:
- any nucleo-f0xx : works
- nucleo-f3xx : needs adaptation in boards/nucleo-f3xx (Makefile.dep and include/periph_conf.h)
- nucleo-f103 : needs adaptation in boards/nucleo-f103 but doesn't build => link error that I don't understand
- nucleo-f4** : couldn't try on-board but builds after adaptation of the boards config
- nucleo-l1 : hard fault handler after initializing the rtc. This is reproducible in master. in #6173 there's no problem.

File: cpu/stm32_common/periph/rtc.c (hunk 1 above)
Comment (vincent-d) [inline cpu/stm32_common/periph/rtc.c]: Fixed

**Comment by aabadie** [comment]
I opened #6462 about the crash with the nucleo-l1 which is already in master.

File: cpu/stm32_common/periph/rtc.c [hunk 2]
```diff
@@ -20,6 +25,10 @@
 #include "periph/rtc.h"
 #include "periph_conf.h"
 
+#if defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F2) || \
```
Comment (aabadie) [inline cpu/stm32_common/periph/rtc.c]: Why not checking the stm32 F1 family ?

**Comment by aabadie** [comment]
- tested on nucleo-f446: works
- re-tested on nucleo-f303: works

the stm32 F1 is not taken into account, why ?

**Comment by vincent-d** [comment]
@aabadie the RTC is different on stm32f1 so this driver won't work

**Comment by aabadie** [comment]
ok, it makes sense. For the nucleo-l1, we have #6462 for keeping track of the problem (not introduced by this PR).
Let me give an ACK then.

**Comment by aabadie** [comment]
and please squash ;)

**Comment by vincent-d** [comment]
Squashed

**Comment by PeterKietzmann** [comment]
@aabadie ACK holds? I see green lights!!!

**Comment by aabadie** [comment]
So go !

**Pull request merged**